app/models/concerns/hostext/search.rb

```diff
@@ -235,6 +235,7 @@ def search_by_os_minor(key, operator, value)
         z ||= 0
         operatingsystem_ids = []
         Operatingsystem.all.find_each do |os|
+          next if os.minor.empty?
```

Comment:
You can pre-filter the empty ones when pulling them from the db
```suggestion
        Operatingsystem.where.not(minor: nil).find_each do |os|
```

Reply:
Unless minor can be `''` empty string, I thing this is better solution :)

Reply:
how about:
```suggestion
        Operatingsystem.where.not(minor: [nil, '']).find_each do |os|
```

Reply:
> Unless minor can be `''` empty string, I thing this is better solution :)

I think we should think about empty strings, but it's a bigger effort. Debian doesn't really have minor versions anymore. It used to really have supported minor versions but now a minor is just a roll up of patches to a release. Also with CentOS there are no supported minors (only the latest). Unless you sync content, you can't really install an older minor version.

tl;dr: the Foreman model is complex, may not match reality anymore, needs discussion with whiteboards and/or beer

Reply:
Yeah, the whole inventory would deserve a brainstorm session, but definitely out of scope of this PR.